code/API_definitions/age-verification.yaml

```
@@ -0,0 +1,511 @@
+openapi: 3.0.3
+info:
+  title: Know Your Customer Age Verification
+
+  description: |
+
+    # Summary
+    This API provides the customer with the ability to check if the user of the line is older than a provided age, in order to provide API customer's age-restricted services, access to its age-restricted website etc..
+
+    # Authorization and authentication
+
+    The "Camara Security and Interoperability Profile" provides details on how a client requests an access token. Please refer to Identify and Consent Management (https://github.com/camaraproject/IdentityAndConsentManagement/) for the released version of the Profile.
+
+    Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
+
+    It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.
+
+    # API functionality
+
+    The API defines one service endpoint:
+
+    - `POST /ageverify`
+
+      Takes the network subscription identifier (e.g. the mobile phone number for a mobile network subscriber) and check the age of the user is older than the age's threshold in the request.
+      If information of the user's identity such as `givenName`, `familyName`, `address` match with information held by the MNO.
+
+
+    ## Inputs
+
+    The endpoint request body is a JSON object with the following parameters:
+    - `phoneNumber`: The network subscription identifier (i.e. the phone number of the subscriber). [Required only in case a 2-legged flow has been agreed between API Provider and API Consumer; otherwise, none]

```

Suggestion to replace this sentence with below based on Huub's comment https://github.com/camaraproject/KnowYourCustomer/pull/50#issuecomment-2484902936 :

- `phoneNumber`: The network subscription identifier (i.e. the phone number of the subscriber). [Required only in case a 2-legged flow has been agreed between API Provider and API Consumer;  otherwise, do not include phoneNumber]

(change to 'otherwise, none)

Thanks,
Toshi

---

wdyt @ToshiWakayama-KDDI ? It sounds more restrictive

```suggestion
    - `phoneNumber`: The network subscription identifier (i.e. the phone number of the subscriber). [Required only in case a 2-legged flow has been agreed between API Provider and API Consumer following CAMARA directives; otherwise, phoneNumber must not be included]

```